plugins/formal-verification/formver.viper/src/org/jetbrains/kotlin/formver/viper/errors/VerificationError.kt

```diff
@@ -6,28 +6,21 @@
 package org.jetbrains.kotlin.formver.viper.errors
 
 import org.jetbrains.kotlin.formver.viper.ast.*
-import viper.silver.verifier.DummyReason.offendingNode
-
-/**
- * This class acts as wrapper for Viper's [viper.silver.verifier.ErrorReason].
- * This is necessary since extension functions on [viper.silver.verifier.ErrorReason] cannot be
- * used outside the class' package.
- */
-data class ErrorReason(private val reason: viper.silver.verifier.ErrorReason) :
-    IntoSilverCallable by delegateToSilverCallable(reason.offendingNode())
 
 class VerificationError private constructor(
     val result: viper.silver.verifier.VerificationError,
-) : VerifierError, IntoSilverCallable by delegateToSilverCallable(offendingNode()) {
+) : VerifierError {
     companion object {
         fun fromSilver(result: viper.silicon.interfaces.VerificationResult): VerificationError {
             check(result.isFatal) { "The verification result must contain an error to be converted." }
             return VerificationError((result as viper.silicon.interfaces.Failure).message())
         }
     }
 
-    val reason: ErrorReason
-        get() = ErrorReason(result.reason())
+    val offendingNode: Node
+        get() = Node(result.offendingNode())
+    val reasonOffendingNode: Node
+        get() = Node(result.reason().offendingNode())
```

Comment:
This is not very descriptive.  Could you document what these correspond to and name them correspondingly?